core/services/relay/evm/types/types.go

```diff
@@ -54,17 +56,59 @@ type CodecConfig struct {
 
 type ChainCodecConfig struct {
 	TypeABI         string                `json:"typeAbi" toml:"typeABI"`
-	ModifierConfigs codec.ModifiersConfig `toml:"modifierConfigs,omitempty"`
+	ModifierConfigs codec.ModifiersConfig `json:"modifierConfigs,omitempty" toml:"modifierConfigs,omitempty"`
+}
+
+type ContractPollingFilter struct {
+	GenericEventNames []string `json:"genericEventNames"`
+	PollingFilter     `json:"pollingFilter"`
+}
+
+type PollingFilter struct {
+	Topic2       evmtypes.HashArray `json:"topic2"`       // list of possible values for topic2
+	Topic3       evmtypes.HashArray `json:"topic3"`       // list of possible values for topic3
+	Topic4       evmtypes.HashArray `json:"topic4"`       // list of possible values for topic4
+	Retention    models.Interval    `json:"retention"`    // maximum amount of time to retain logs
```

Comment:
Something seems off here about the `models.Interval` type.  If I'm understanding this correctly, we're using a postgresql type to read the retention time from the job spec, but then when we actually need to send it to the db, we convert from the postgres type into native golang type `time.Duration`.

If we're going to have two different types, shouldn't it be the other way around?

Reply:
Admittedly, I never knew about this `models.Interval` type... if the intention is to encourage LogPoller to change to that type as well soon, I could see that making sense.

Reply:
time.Duration lacks JsonMarshall/JsonUnmarshall